### ISSUE TITLE
bad_channel_flag repeats idefinately

### ISSUE THREAD

**Issue by rfshipman** [craigkulesa/gusto-datasystem#17]
It is good that the add band channel flag checks if it has been done already.  The way it is implemented it will to the beginning of the scans when done.  

I'm using it to update every FITS file at level 0.8 so I run
python set_bad_chflag.py "ACS3*"

This runs over all files in directory for ACS3 and then continues from the beginning.  I have to ctl-C out.

**Comment by abegyoung** [comment]
I can't reproduce this behavior.

**Comment by rfshipman** [comment]
Note that the 2nd time through the HISTORY entry was present so nothing changed.  The script was opening all the fits files with ACS3* in the name.   Maybe there is a way to call the task so that it ends properly.  
I would consider this a low priority issue.

**Comment by abegyoung** [comment]
Ok, so google AI tells me glob.glob( ) will sometimes get duplicates.  Use of the recursive ** will do it for globbing through directories.  set( ) will fix this.

More likely, since the level0.9 directory structure was in two directories, ACS3/ and ACS5/, I iterated the glob over the B1 and B2 paths.  If your config.ini has B1 and B2 paths the same directory, you'll get 2X through the same files as the glob was done twice on the same path.

This still doesn't cause an indefinite loop.  I've restructured the config.ini to keep up with the current level0.9 single directory structure now.  Hopefully this was the cause of your issue.

cheers
